_includes/locale/Locale.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ */
+
+  declare(strict_types=1);
+
+  namespace Keyman\Site\com\keyman;
+
+  class Locale {
+    public const DEFAULT_LOCALE = 'en';
+
+    public const CROWDIN_LOCALES = array(
+      'en',
+      'es-ES',
+      'fr-FR'
+    );
+
+    // xx-YY locale as specified in crowdin %locale%
+    private static $currentLocale = Locale::DEFAULT_LOCALE;
+
+    // If set to true, warn about untranslated strings
+    private static $debugMode = false; // Keep false in production
+
+    // Associative array to hold translated strings in "current locale" and English
+    private static $langArray = [];
+    private static $langArrayEn = [];
+
+    /**
+     * Return the current locale. Fallback to 'en'
+     * @return $currentLocale
+     */
+    public static function currentLocale() {
+      return Locale::$currentLocale;
+    }
+
+    /**
+     * Validate and override the current locale
+     * @param $locale - the new current locale (xx-YY as specified in crowdin %locale%)
+     */
+    public static function overrideCurrentLocale($locale) {
```

Comment:
`overrideCurrentLocale` should probably be `setLocale`. This is a global setting for the site. Here, we set the primary locale, with an array of fallbacks, ending always in 'en'.

```php
function setLocale($locale) {
  // TODO: support other fallbacks such as es-419->es
  // for now, just support locale and 'en'
  Locale::$currentLocales = [$locale, 'en'];
}
```

So we don't do validation here, because we'll always have fallback anyway.